src/main/java/seedu/mtracker/model/Instrument.java

```diff
@@ -77,6 +83,17 @@ public void editName(HashMap<String, String> editedParameters) {
         setName(editedParameters.get(NAME_ATTRIBUTE));
     }
 
+    public void editDoneStatus(HashMap<String, String> editedParameters) {
+        if (!editedParameters.containsKey(DONE_ATTRIBUTE)) {
+            return;
+        }
+        if (editedParameters.get(DONE_ATTRIBUTE).equals(DONE_INDICATOR)) {
+            markAsDone();
+        } else {

```

Comment:
can add an assert in the else block that editedParameters.get(DONE_ATTRIBUTE) equals "undone" only